File: app/src/main/java/com/khalid/interviewcracker/util/AndroidFileOpener.java
package com.khalid.interviewcracker.util;

import java.io.IOException;
import java.io.InputStream;

import android.content.Context;

import com.khalid.interviewcracker.dispatcher.FileOpener;


public class AndroidFileOpener implements FileOpener {

	protected static Context mContext;

	public AndroidFileOpener(Context context) {
		mContext = context;
	}

	@Override
	public InputStream openFile(String filename) throws IOException {
		return mContext.getResources().getAssets().open(filename);
	}
}

File: app/src/main/java/com/khalid/interviewcracker/util/AndroidFileOpener.kt
package com.khalid.interviewcracker.util

import java.io.IOException
import java.io.InputStream

import android.content.Context
import com.khalid.interviewcracker.ICApplication

import com.khalid.interviewcracker.dispatcher.FileOpener
import javax.inject.Inject


class AndroidFileOpener : FileOpener {

   @Inject lateinit var context:Context

    init {
        ICApplication.appComponent.inject(this)
    }

    @Throws(IOException::class)
    override fun openFile(filename: String): InputStream {
        return context.resources.assets.open(filename)
    }

}
